sensor/common/internalmessage/internal_message_pubsub.go

```diff
@@ -0,0 +1,80 @@
+package internalmessage
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	SensorMessageSoftRestart = iota
+)
+
+// SensorInternalMessageCallback is the callback used by subscribers.
+type SensorInternalMessageCallback func(message *SensorInternalMessage)
```

Comment:
Based on this call later: `go subCallback(msg)`, I think, that maybe we should add here a context to have a possibility to cancel the goroutine?

```suggestion
type SensorInternalMessageCallback func(ctx context.Context, message *SensorInternalMessage)
```

Reply:
I like this suggestion but I would maybe use the interface `concurrency.Waitable` that way you can pass a context or a signal to the cancel the goroutine.

Reply:
@vikin91 `SensorInternalMessage` has a context which is used exactly for that (see helper function `IsExpired`). 

From `sensor@275`:
```go
	err = s.pubSub.Subscribe(internalmessage.SensorMessageSoftRestart, func(message *internalmessage.SensorInternalMessage) {
		if message.IsExpired() {
			return
		}
		s.centralCommunicationLock.Lock()
		defer s.centralCommunicationLock.Unlock()
		if s.centralCommunication == nil {
			log.Warnf("Sensor connection was not yet established when internal message for connection restart was received. Skipping soft restart")
			return
		}
		s.centralCommunication.Stop(errors.Wrap(errForcedConnectionRestart, message.Text))
	})
```

Does that solve the problem you pointed out?

Reply:
Resolving